paasta_tools/cli/cmds/spark_run.py

```diff
@@ -522,44 +525,29 @@ def create_spark_config_str(spark_config_dict, is_mrjob):
 
 
 def emit_resource_requirements(spark_config_dict, paasta_cluster, webui_url):
-    num_executors = int(spark_config_dict["spark.cores.max"]) / int(
-        spark_config_dict["spark.executor.cores"]
-    )
-    memory_per_executor = calculate_memory_per_executor(
-        spark_config_dict["spark.executor.memory"],
-        spark_config_dict.get("spark.mesos.executor.memoryOverhead"),
-    )
-
-    desired_resources = {
-        "cpus": int(spark_config_dict["spark.cores.max"]),
-        "mem": memory_per_executor * num_executors,
-        "disk": memory_per_executor
-        * num_executors,  # rough guess since spark does not collect this information
-    }
-    dimensions = {
-        "framework_name": spark_config_dict["spark.app.name"],
-        "webui_url": webui_url,
-    }
+    paasta_print("Sending resource request metrics to Clusterman")
 
+    desired_resources = get_spark_resource_requirements(spark_config_dict, webui_url)
     constraints = parse_constraints_string(spark_config_dict["spark.mesos.constraints"])
     pool = constraints["pool"]
 
-    paasta_print("Sending resource request metrics to Clusterman")
     aws_region = get_aws_region_for_paasta_cluster(paasta_cluster)
     metrics_client = clusterman_metrics.ClustermanMetricsBotoClient(
         region_name=aws_region, app_identifier=pool
     )
 
-    estimated_cost = clusterman_metrics.util.costs.estimate_cost_per_hour(
-        cluster=paasta_cluster,
-        pool=pool,
-        cpus=desired_resources["cpus"],
-        mem=desired_resources["mem"],
-    )
-    message = "Resource request ({} cpus and {} MB memory total) is estimated to cost ${} per hour".format(
-        desired_resources["cpus"], desired_resources["mem"], estimated_cost
+    cpus, mem = None, None
+    for key, value in desired_resources.items():
+        if "requested_cpus" in key:
+            cpus = value
+        elif "requested_mem" in key:
+            mem = value
```

Comment:
Maybe you should have `get_spark_resource_requirements` just returns the resources (rather than the full metric names with dimensions) so you don't have to do this.  Kind of feel like you're working around your own code.

You could maybe even have a separate method to generate the dimensions so that that's also easily available?

Reply:
I don't feel like any solution here is _great_, but I came up with something that's maybe a little better.  Thoughts?